lib/src/contributors/class/from_json_assist_contributor/from_json_assist_contributor.dart

```diff
@@ -99,16 +99,30 @@ class FromJsonAssistContributor extends Object
     required final ClassElement classElement,
     required final DartEditBuilder builder,
   }) {
+    final String className = classElement.name;
+    final ConstructorElement? defaultConstructor = classElement.defaultConstructor;
+    final bool isConst = defaultConstructor?.isConst ?? true;
+    final List<VariableElement> fields = <VariableElement>[
+      ...classElement.dataClassFinalFields,
+      ...classElement.chainSuperClassDataClassFinalFields,
+    ];
+
     builder
       ..writeln()
-      ..writeln('/// Creates an instance of [${classElement.name}] from [json]')
-      ..writeln('factory ${classElement.name}.fromJson(Map<String, dynamic> json) {')
-      ..writeln('return ${classElement.name}(');
+      ..writeln('/// Creates an instance of [$className] from [json]')
+      ..writeln('factory $className.fromJson(Map<String, dynamic> json) {');
+
+    if (isConst && fields.isEmpty) {
+      builder
```

Comment:
`fromJson` can use the const constructor as it's ok to get the same instance

Reply:
That was my bad when naming the issue

I have a different issue for this one
https://github.com/spideythewebhead/data_class_plugin/issues/104